modules/settings.js

```diff
@@ -231,44 +241,160 @@ class Settings {
+                const tmpPath = path.join(this.userHomePath, network.dataDir, network.ipcFile);
+                if (fs.existsSync(tmpPath)) ipcPaths.push(tmpPath);
+            });
+
+            if (_.isEmpty(ipcPaths)) {
+                // set IPC endpoint path for own node (read network from config)
+                ipcPaths.push(path.join(this.userHomePath,
+                    networks[this.network].dataDir,
+                    networks[this.network].ipcFile
+                ));
+                this._ownNode = true;
+            }
         }
+        this._log.debug(`IPC path: ${ipcPaths}`);
 
-        this._log.debug(`IPC path: ${ipcPath}`);
+        return ipcPaths[0];
+    }
 
-        return ipcPath;
+    get nodeOptions() {
+        return argv.nodeOptions;
     }
 
     get nodeType() {
-        return argv.node;
+        if (argv.node) return argv.node;
+
+        const json = this.clientBinariesJSON;
+
+        _.keys(json.clients).forEach((client) => {  // eslint-disable-line consistent-return
+            if (client === this.loadConfig('node.type')) return client;
+        });
+
+        return json.defaults.nodeType;
+    }
+
+    set nodeType(type) {
+        this.saveConfig('node.type', type);
     }
 
     get network() {
-        return argv.network;
+        if (argv.network) return argv.network;
+
+        const json = this.clientBinariesJSON.clients;
+        const networks = json[this.nodeType].platforms[this.platform][process.arch].networks;
+        _.keys(networks).forEach((network) => {  // eslint-disable-line consistent-return
+            if (network === this.loadConfig('node.network')) return network;
+        });
+
+        return _.keys(networks)[0];
     }
 
-    get nodeOptions() {
-        return argv.nodeOptions;
+    set network(network) {
+        this.saveConfig('node.network', network);
+    }
+
+    get language() {
+        return this.loadConfig('i18n');
+    }
+
+    set language(langCode) {
+        this.saveConfig('i18n', langCode);
+    }
+
+    initConfig() {
+        global.config.insert({
+            i18n: i18n.getBestMatchedLangCode(app.getLocale())
+        });
     }
 
-    loadUserData(path) {
-        const fullPath = this.constructUserDataPath(path);
+    saveConfig(key, value) {
+        let obj = global.config.get(1);
+
+        if (!obj) {
+            this.initConfig();
+            obj = global.config.get(1);
+        }
+
+        if (this.deepEval(obj, key) !== value) {
+            this.deepEval(obj, key, value);
+            global.config.update(obj);
+
+            this._log.debug(`Settings: saveConfig('${key}', '${value}')`);
+            this._log.trace(global.config.data);
+        }
+    }
+
+    loadConfig(key) {
+        const obj = global.config.get(1);
+
+        if (!obj) {
+            this.initConfig();
+            return this.loadConfig(key);
+        }
+
+        return this.deepEval(obj, key);
+    }
+
+    deepEval(obj, is, value) {
```

Comment:
hehe, I just figured that this has the same functionality as [lodash.set](https://lodash.com/docs/#set) and  [lodash.get](https://lodash.com/docs/#get), which is available as https://www.npmjs.com/package/lodash-get-or-set. 
I will change that in hopes of stricter and more reliable validation.